### PR TITLE
Global Styles: Add support for additional text settings

### PR DESCRIPTION
**Pull request by danieldudzic**
This PR:

- Adds necessary utility functions to handle: `text_decoration`, `text_transform` and `letter_spacing`.
- Orders all the utility functions alphabetically.
- Fixes `fontFamily` prop handling.

#### User Facing Testing

1. You will also need this branch to test: https://github.com/woocommerce/woocommerce-blocks/tree/update/8087-sale-badge-global-styles-options
2. Add the `Products` and the `On-Sale Badge` Blocks on a new page.
3. Make sure typography settings ( `Font Family`, `Appearance`, `Letter spacing`, `Letter case`, and `Decoration` ) are available and working correctly both in the frontend and in the editor.

* [x] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### CODE REVIEW

**Comment by danieldudzic** [comment]
Note: Since I have ordered functions alphabetically in `src/Utils/StyleAttributesUtils.php` the actual changes are showing really weirdly in Github.

File: assets/js/base/hooks/use-typography-props.ts [hunk 1]
```diff
@@ -2,30 +2,53 @@
...
+
 type WithStyle = {
 	style: Record< string, unknown >;
 };
 
-export const useTypographyProps = ( attributes: unknown ): WithStyle => {
+export const useTypographyProps = (
+	attributes: unknown
+): WithStyle & WithClass => {
+	if ( ! isFeaturePluginBuild() ) {
```
Comment (Aljullu) [inline assets/js/base/hooks/use-typography-props.ts]: We need to support font size and line height in WC core, as they are not experimental.

Thinking about this, IMO `useTypographyProps()` shouldn't check whether it's WC core or the feature plugin. That should be checked at the block level when defining which styles are supported. What do you think?

**Review by Aljullu — COMMENTED** [review]
Tested with #8334 and it works as expected. However, I left some comments below: :point_down: 

> Note: Since I have ordered functions alphabetically in `src/Utils/StyleAttributesUtils.php` the actual changes are showing really weirdly in Github.

When reordering things in code it's usually a good idea to do one commit for the actual changes and, in a separate commit, reorder the code. This way, the code can be reviewed commit-by-commit. :slightly_smiling_face: 

Code changes look good, and it makes sense to me to order the style functions alphabetically. However:
* there are some 'utils' functions which I think should be at the bottom: `get_classes_and_styles_by_attributes`, `get_classes_by_attributes`, `get_styles_by_attributes`, `get_preset_value` and `get_color_value`. They are currently in between the other functions but without any specific order, apparently.
* there are some functions which don't seem to be ordered yet (ie: `get_border_width_class_and_style` is before `get_align_class_and_style`).

File: assets/js/base/hooks/use-typography-props.ts (hunk 1 above)
Comment (danieldudzic) [reply assets/js/base/hooks/use-typography-props.ts]: Right. Good point. I agree. I'll remove the check. I'll also check and see if all blocks utilizing `useTypographyProps()` include the necessary check.

**Comment by danieldudzic** [comment]
> When reordering things in code it's usually a good idea to do one commit for the actual changes and, in a separate commit, reorder the code. This way, the code can be reviewed commit-by-commit. 🙂

Right! Makes perfect sense. Would you like me to re-do the commits for this PR?

> there are some 'utils' functions which I think should be at the bottom: get_classes_and_styles_by_attributes, get_classes_by_attributes, get_styles_by_attributes, get_preset_value and get_color_value. They are currently in between the other functions but without any specific order, apparently.

I wanted to separate the `get_classes_and_styles_by_attributes()` functions from the util functions, but perhaps I didn't get it quite right. I'll have another look.

> there are some functions which don't seem to be ordered yet (ie: get_border_width_class_and_style is before get_align_class_and_style).

Good catch. I'll re-review.

**Comment by Aljullu** [comment]
> Right! Makes perfect sense. Would you like me to re-do the commits for this PR?

No need to do it for this PR. Just an advise for future ones. :slightly_smiling_face: 

> I wanted to separate the `get_classes_and_styles_by_attributes()` functions from the util functions, but perhaps I didn't get it quite right. I'll have another look.

Oh right, that one if different from the rest of the 'util' functions. It makes sense. Sorry for the confusion!

**Comment by danieldudzic** [comment]
I have now ordered the functions in the following way:
- Utility functions - top
- Class and style functions - middle
- Main class and style function - bottom

**Review by Aljullu — APPROVED** [review]
> I have now ordered the functions in the following way:

It makes a lot of sense, thanks for following up on this.

LGTM :ship:

**Comment by danieldudzic** [comment]
Failing tests are unrelated. I'm going to merge.

**Pull request merged**